packages/plugin-ext/src/main/browser/plugin-contribution-handler.ts

```diff
@@ -300,7 +300,7 @@ export class PluginContributionHandler {
         }
 
         if (contributions.colors) {
-            pushContribution('colors', () => this.colors.register(...contributions.colors));
+            pushContribution('colors', () => this.colors.register(...contributions.colors!));
```

Comment:
Would it be safer to do this:
```typescript
const {colors} = contributions;
pushContribution('colors', () => this.colors.register(...colors));
```
Then TS is happy that colors will exist without the non-null assertion.

Reply:
I feel like it is OK with the way it currently is, not a fan of binding consts just to please typing...

Reply:
It isn't _really_ just to please typing in this case. It seems reasonable to believe that nothing will delete the `colors` between this call and when the callback is called, but TS is reminding us that `delete` is a thing, and the rest of this PR shows we're not afraid to use it.

Reply:
> It seems reasonable to believe that nothing will delete the colors between this call and when the callback is called [...]

It's the goal of the current implementation.

> [...] but TS is reminding us that `delete` is a thing, and the rest of this PR shows we're not afraid to use it.

Agreed, but once again in this specific instance, adding the `non-null` coercion or capturing the value out of the `contribution` object wouldn't help much should a `delete` be introduced for any reason. In one case it will fail because of an undefined value that was supposed to be defined, and in the other case it will keep working despite the value being deleted.

TL;DR: whenever we have code that isn't specifically tailored like this one, we could do as you say and capture properties to ensure they don't disappear from the object. Although ultimately removing properties mid-flight sounds like a foot gun.